18F/identity-idp
# LG-14382: Update intl-tel-input to the latest version

## 🎫 Ticket

[LG-14382](https://cm-jira.usa.gov/browse/LG-14382)

## 🛠 Summary of changes

Updates `intl-tel-input` to the latest version, from 17.0.19 to 24.5.0.

Release notes: https://github.com/jackocnr/intl-tel-input/releases

This is a significant upgrade spanning multiple major versions, and had more than a handful of breaking changes to sift through.

While it's a maintenance task, it's also partly motivated by discovery in #11189 that the phone input had accessibility issues uncovered by the changes there. Newer versions of the library address those issues. 

There's also at least one noteworthy feature included in newer versions: the ability to search for countries in the dropdown.

**Draft:** Working to resolve an issue with "Single Country Selection" (not currently used in the application).

## 📜 Testing Plan

Verify there's no regression in behavior of the phone input

1. Go to http://localhost:3000/components/preview/phone_input/preview
2. Interact with the phone fields
3. Go to https://idp.dev.identitysandbox.gov/components/preview/phone_input/preview
4. Interact with the phone fields
5. Observe that they behave largely the same (aside from new features in latest version, like country search)

You might also want to check other usage in the application for regressions:

- Adding a phone as an MFA method
- Hybrid document capture handoff
- Phone verification step in identity verification

## 👀 Screenshots

Before|After
---|---
![image](https://github.com/user-attachments/assets/d47a5da0-dbd8-4d82-9807-494f85977479)|![image](https://github.com/user-attachments/assets/2f932c9d-812b-4d09-909d-f961927e8358)
